### DISCUSSION TITLE
Loose user custom column data on publish

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#10743] [Help] [answered]
I added some users meta-data using custom columns, in the BB user internal table.
Each time I publish my app, the live meta-data are erased and replaced with the values of the dev environment.
I use version V2.6.11
CC @mielune

**Comment by melohagan** [comment]
Hey @NicolasFabre81 

Whenever you publish your app data is synced from the dev environment to the production app.

**Reply by NicolasFabre81** [reply]
Hello @melohagan ,
In my apps I always use postgreSQL as data source. I only have to use bb internal tables for the user table.
Of course my dev database end prod database are completely independent.

I don't understand how one can work if internal tables rows are synced from dev to prod ?

It would mean you loose data each time you publish ?

**Reply by melohagan** [reply]
Data should continuously be synced from Prod to Dev, for example when you reload the page of the *Data** section.

Dev data is only synced to Prod whenever you click **Publish**. 

There may possibly be an issue in that the user data is not being synced from Prod into Dev correctly.

**Reply by NicolasFabre81** [reply, marked as answer]
I think you pointed the issue. But I just made a test and it worked ! Maybe it happens on particular cases... I'll let you know if I manage to reproduce it.

**Reply by melohagan** [reply]
Good to hear, definitely do raise a bug if you can reproduce!
